analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,7 +125,10 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
+    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, SyntaxTree tree) =>
+        !IsUnchanged(tree) && IsInSonarLintContextAndIncluded(sonarLintXml, tree.FilePath);
+
+    private bool IsInSonarLintContextAndIncluded(SonarLintXmlReader sonarLintXml, string filePath) =>
         ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
         || (FileInclusionCache.GetValue(Compilation, _ => new()) is var cache
             && cache.GetOrAdd(filePath, _ => IsFileIncluded(sonarLintXml, filePath)));
```

Comment:
Extracting these two lines to `IsFileIncluded(sonarLintXml, filePath)` could help with organizing the rest in more readable way... I'm not sure

Reply:
I deleted `ShouldAnalyzeFile` and renamed `IsInSonarLintContextAndIncluded`.

I won't touch here `IsFileIncluded` as in next PR I transfer it to `SonarLintXmlReader`.